### PR TITLE
Add `SingleChildScrollView` for `NavigationRail`

### LINKED ISSUE #89167: Making NavigationRail scrollable
## Use case

By default, when a `NavigationRailDestination` inside a `NavigationRail` goes out of the viewport (e.g. the window is resized to be less than the destinations), it will overflow and cause the infamous yellow-and-black-striped error to occur. One way to fix this is to make the `NavigationRail` itself scrollable. Currently, the only way to do this is rather janky.

**Error-causing:**

```dart
Row(
  children: [
    NavigationRail(...),
    ...
  ]
)
```

**The "fix" that makes it scrollable:**

```dart
Row(
  children: [
    LayoutBuilder(
      builder: (context, constraints) {
        return SingleChildScrollView(
          child: ConstrainedBox(
            constraints: BoxConstraints(
              minHeight: constraints.maxHeight,
            ),
            child: IntrinsicHeight(
              child: NavigationRail(...)
            )
          )
        );
      }
    ),
    ...
  ]
)
```

This fix is mentioned in [this StackOverflow answer](https://stackoverflow.com/a/61876930/14021404) and is also used by [Flutter Gallery](https://github.com/flutter/gallery/blob/860b616cea37140e55ee23e92b2e4309ac9a698c/lib/studies/reply/adaptive_nav.dart#L179-L240). However, it is janky and cluttered. It even uses an `IntrinsicHeight` wherein the docs explicitly state that:

_In the [class constructor](https://api.flutter.dev/flutter/widgets/IntrinsicHeight/IntrinsicHeight.html):_
> This class is relatively expensive. Avoid using it where possible.

_In the [class itself](https://api.flutter.dev/flutter/widgets/IntrinsicHeight-class.html):_
> This class is relatively expensive, because it adds a speculative layout pass before the final layout phase. Avoid using it where possible. In the worst case, this widget can result in a layout that is O(N²) in the depth of the tree.

I want to avoid using widgets that the docs itself says not to use. However, I cannot find any other way to make it scrollable to avoid the infamous yellow-and-black-striped error.

## Proposal
I propose a simple solution to this problem:
```dart
Row(
  children: [
    NavigationRail(
      scrollable: true,
      ...
    ),
    ...
  ]
)
```

Not only does this makes things easier and intuitive, but is also easily customizable and less clutter.

### PR DESCRIPTION
**Pull request by mbfakourii**
## Description
Add `SingleChildScrollView` to `NavigationRail` for scrolling.
Closes: #89167

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by goderbauer** [comment]
Hey @mbfakourii, can you please add a test to this to make sure we never regress this? See the comment posted by the [flutter-dashboard](https://github.com/apps/flutter-dashboard). Also, it looks like this change is failing some of the checks. Can you please take a look at that and address it? Thanks!

**Comment by mbfakourii** [comment]
> Hey @mbfakourii, can you please add a test to this to make sure we never regress this? See the comment posted by the [flutter-dashboard](https://github.com/apps/flutter-dashboard). Also, it looks like this change is failing some of the checks. Can you please take a look at that and address it? Thanks!

Tests were added

**Comment by wjkoh** [comment]
This is fantastic! I hope the PR is approved quickly.

**Comment by goderbauer** [comment]
Looks like some existing tests are failing, can you please take a look and fix those up?

**Comment by mbfakourii** [comment]
> Looks like some existing tests are failing, can you please take a look and fix those up?

I checked the fail tests, please check [here](https://github.com/flutter/flutter/pull/137415#issuecomment-1793400346).

**Comment by goderbauer** [comment]
There's at least one failing test that's pretty clearly linked to the change here:

```dart
07:58 +8848 ~22: /b/s/w/ir/x/w/flutter/packages/flutter/test/material/navigation_rail_test.dart: Semantics - labelType=[none]                                                                          
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure was thrown running a test:
Expected: semantics node matching:
          TestSemantics(
            id: 0,
            rect: Rect.fromLTRB(0.0, 0.0, 2400.0, 1800.0),
            elevation: 0.0,
            thickness: 0.0,
            children: <TestSemantics>[
              TestSemantics(
                textDirection: TextDirection.ltr,
                children: <TestSemantics>[
                  TestSemantics(
                    children: <TestSemantics>[
                      TestSemantics(
                        flags: <SemanticsFlag>[SemanticsFlag.scopesRoute],
                        children: <TestSemantics>[
                          TestSemantics(
                            flags: <SemanticsFlag>[SemanticsFlag.isSelected,
SemanticsFlag.isFocusable],
                            actions: <SemanticsAction>[SemanticsAction.tap],
                            label: 'Abc
          Tab 1 of 4',
                            textDirection: TextDirection.ltr,
                            children: <TestSemantics>[
                            ],
                          ),
                          TestSemantics(
                            flags: <SemanticsFlag>[SemanticsFlag.isFocusable],
                            actions: <SemanticsAction>[SemanticsAction.tap],
                            label: 'Def
          Tab 2 of 4',
                            textDirection: TextDirection.ltr,
                            children: <TestSemantics>[
                            ],
                          ),
                          TestSemantics(
                            flags: <SemanticsFlag>[SemanticsFlag.isFocusable],
                            actions: <SemanticsAction>[SemanticsAction.tap],
                            label: 'Ghi
          Tab 3 of 4',
                            textDirection: TextDirection.ltr,
                            children: <TestSemantics>[
                            ],
                          ),
                          TestSemantics(
                            flags: <SemanticsFlag>[SemanticsFlag.isFocusable],
                            actions: <SemanticsAction>[SemanticsAction.tap],
                            label: 'Jkl
          Tab 4 of 4',
                            textDirection: TextDirection.ltr,
                            children: <TestSemantics>[
                            ],
                          ),
                          TestSemantics(
                            label: 'body',
                            textDirection: TextDirection.ltr,
                            children: <TestSemantics>[
                            ],
                          ),
                        ],
                      ),
                    ],
                  ),
                ],
              ),
            ],
          )
  Actual: SemanticsTester:<SemanticsTester for SemanticsNode#0(Rect.fromLTRB(0.0, 0.0, 2400.0,
1800.0))>
   Which: expected node id null to have 5 children but found 2.
          Current SemanticsNode tree:
            SemanticsNode#0
             │ Rect.fromLTRB(0.0, 0.0, 2400.0, 1800.0)
             │
             └─SemanticsNode#1
               │ Rect.fromLTRB(0.0, 0.0, 800.0, 600.0) scaled by 3.0x
               │ textDirection: ltr
               │
               └─SemanticsNode#2
                 │ Rect.fromLTRB(0.0, 0.0, 800.0, 600.0)
                 │ sortKey: OrdinalSortKey#1e145(order: 0.0)
                 │
                 └─SemanticsNode#3
                   │ Rect.fromLTRB(0.0, 0.0, 800.0, 600.0)
                   │ flags: scopesRoute
                   │
                   ├─SemanticsNode#4
                   │ │ Rect.fromLTRB(0.0, 8.0, 80.0, 184.0)
                   │ │ flags: hasImplicitScrolling
                   │ │ scrollExtentMin: 0.0
                   │ │ scrollPosition: 0.0
                   │ │ scrollExtentMax: 0.0
                   │ │
                   │ ├─SemanticsNode#5
                   │ │   Rect.fromLTRB(0.0, 0.0, 80.0, 44.0)
                   │ │   actions: tap
                   │ │   flags: isSelected, isFocusable
                   │ │   label:
                   │ │     "Abc
                   │ │     Tab 1 of 4"
                   │ │   textDirection: ltr
                   │ │
                   │ ├─SemanticsNode#6
                   │ │   Rect.fromLTRB(0.0, 44.0, 80.0, 88.0)
                   │ │   actions: tap
                   │ │   flags: isFocusable
                   │ │   label:
                   │ │     "Def
                   │ │     Tab 2 of 4"
                   │ │   textDirection: ltr
                   │ │
                   │ ├─SemanticsNode#7
                   │ │   Rect.fromLTRB(0.0, 88.0, 80.0, 132.0)
                   │ │   actions: tap
                   │ │   flags: isFocusable
                   │ │   label:
                   │ │     "Ghi
                   │ │     Tab 3 of 4"
                   │ │   textDirection: ltr
                   │ │
                   │ └─SemanticsNode#8
                   │     Rect.fromLTRB(0.0, 132.0, 80.0, 176.0)
                   │     actions: tap
                   │     flags: isFocusable
                   │     label:
                   │       "Jkl
                   │       Tab 4 of 4"
                   │     textDirection: ltr
                   │
                   └─SemanticsNode#9
                       Rect.fromLTRB(80.0, 290.0, 800.0, 310.0)
                       label: "body"
                       textDirection: ltr
          The semantics tree would have matched the following configuration:
            TestSemantics.root(
              children: <TestSemantics>[
                TestSemantics(
                  id: 1,
                  textDirection: TextDirection.ltr,
                  children: <TestSemantics>[
                    TestSemantics(
                      id: 2,
                      children: <TestSemantics>[
                        TestSemantics(
                          id: 3,
                          flags: <SemanticsFlag>[SemanticsFlag.scopesRoute],
                          children: <TestSemantics>[
                            TestSemantics(
                              id: 4,
                              flags: <SemanticsFlag>[SemanticsFlag.hasImplicitScrolling],
                              children: <TestSemantics>[
                                TestSemantics(
                                  id: 5,
                                  flags: <SemanticsFlag>[SemanticsFlag.isSelected,
SemanticsFlag.isFocusable],
                                  actions: <SemanticsAction>[SemanticsAction.tap],
                                  label: 'Abc\nTab 1 of 4',
                                  textDirection: TextDirection.ltr,
                                ),
                                TestSemantics(
                                  id: 6,
                                  flags: <SemanticsFlag>[SemanticsFlag.isFocusable],
                                  actions: <SemanticsAction>[SemanticsAction.tap],
                                  label: 'Def\nTab 2 of 4',
                                  textDirection: TextDirection.ltr,
                                ),
                                TestSemantics(
                                  id: 7,
                                  flags: <SemanticsFlag>[SemanticsFlag.isFocusable],
                                  actions: <SemanticsAction>[SemanticsAction.tap],
                                  label: 'Ghi\nTab 3 of 4',
                                  textDirection: TextDirection.ltr,
                                ),
                                TestSemantics(
                                  id: 8,
                                  flags: <SemanticsFlag>[SemanticsFlag.isFocusable],
                                  actions: <SemanticsAction>[SemanticsAction.tap],
                                  label: 'Jkl\nTab 4 of 4',
                                  textDirection: TextDirection.ltr,
                                ),
                              ],
                            ),
                            TestSemantics(
                              id: 9,
                              label: 'body',
                              textDirection: TextDirection.ltr,
                            ),
                          ],
                        ),
                      ],
                    ),
                  ],
                ),
              ],
            )

When the exception was thrown, this was the stack:
#4      main.<anonymous closure> (file:///b/s/w/ir/x/w/flutter/packages/flutter/test/material/navigation_rail_test.dart:2276:5)
<asynchronous suspension>
#5      testWidgetsWithLeakTracking.wrappedCallBack (package:leak_tracker_flutter_testing/src/test_widgets.dart:108:5)
<asynchronous suspension>
#6      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:168:15)
<asynchronous suspension>
#7      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:1017:5)
<asynchronous suspension>
<asynchronous suspension>
(elided one frame from package:stack_trace)

This was caught by the test expectation on the following line:
  file:///b/s/w/ir/x/w/flutter/packages/flutter/test/material/navigation_rail_test.dart line 2276
The test description was:
  Semantics - labelType=[none]
════════════════════════════════════════════════════════════════════════════════════════════════════

07:58 +8848 ~22 -1: /b/s/w/ir/x/w/flutter/packages/flutter/test/material/navigation_rail_test.dart: Semantics - labelType=[none] [E]                                                                   
  Test failed. See exception logs above.
  The test description was: Semantics - labelType=[none]
```

(from https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8761948290894190433/+/u/run_test.dart_for_framework_tests_shard_and_subshard_libraries/test_stdout).

That one will likely need to get fixed. If you see other unrelated test failures, try rebasing this PR with the latest master and see if they go away. The tree is currently green, so there are no unrelated existing failures.

**Comment by mbfakourii** [comment]
> There's at least one failing test that's pretty clearly linked to the change here:
> 
> ```dart
> 07:58 +8848 ~22: /b/s/w/ir/x/w/flutter/packages/flutter/test/material/navigation_rail_test.dart: Semantics - labelType=[none]                                                                          
> ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
> The following TestFailure was thrown running a test:
> Expected: semantics node matching:
>           TestSemantics(
>             id: 0,
>             rect: Rect.fromLTRB(0.0, 0.0, 2400.0, 1800.0),
>             elevation: 0.0,
>             thickness: 0.0,
>             children: <TestSemantics>[
>               TestSemantics(
>                 textDirection: TextDirection.ltr,
>                 children: <TestSemantics>[
>                   TestSemantics(
>                     children: <TestSemantics>[
>                       TestSemantics(
>                         flags: <SemanticsFlag>[SemanticsFlag.scopesRoute],
>                         children: <TestSemantics>[
>                           TestSemantics(
>                             flags: <SemanticsFlag>[SemanticsFlag.isSelected,
> SemanticsFlag.isFocusable],
>                             actions: <SemanticsAction>[SemanticsAction.tap],
>                             label: 'Abc
>           Tab 1 of 4',
>                             textDirection: TextDirection.ltr,
>                             children: <TestSemantics>[
>                             ],
>                           ),
>                           TestSemantics(
>                             flags: <SemanticsFlag>[SemanticsFlag.isFocusable],
>                             actions: <SemanticsAction>[SemanticsAction.tap],
>                             label: 'Def
>           Tab 2 of 4',
>                             textDirection: TextDirection.ltr,
>                             children: <TestSemantics>[
>                             ],
>                           ),
>                           TestSemantics(
>                             flags: <SemanticsFlag>[SemanticsFlag.isFocusable],
>                             actions: <SemanticsAction>[SemanticsAction.tap],
>                             label: 'Ghi
>           Tab 3 of 4',
>                             textDirection: TextDirection.ltr,
>                             children: <TestSemantics>[
>                             ],
>                           ),
>                           TestSemantics(
>                             flags: <SemanticsFlag>[SemanticsFlag.isFocusable],
>                             actions: <SemanticsAction>[SemanticsAction.tap],
>                             label: 'Jkl
>           Tab 4 of 4',
>                             textDirection: TextDirection.ltr,
>                             children: <TestSemantics>[
>                             ],
>                           ),
>                           TestSemantics(
>                             label: 'body',
>                             textDirection: TextDirection.ltr,
>                             children: <TestSemantics>[
>                             ],
>                           ),
>                         ],
>                       ),
>                     ],
>                   ),
>                 ],
>               ),
>             ],
>           )
>   Actual: SemanticsTester:<SemanticsTester for SemanticsNode#0(Rect.fromLTRB(0.0, 0.0, 2400.0,
> 1800.0))>
>    Which: expected node id null to have 5 children but found 2.
>           Current SemanticsNode tree:
>             SemanticsNode#0
>              │ Rect.fromLTRB(0.0, 0.0, 2400.0, 1800.0)
>              │
>              └─SemanticsNode#1
>                │ Rect.fromLTRB(0.0, 0.0, 800.0, 600.0) scaled by 3.0x
>                │ textDirection: ltr
>                │
>                └─SemanticsNode#2
>                  │ Rect.fromLTRB(0.0, 0.0, 800.0, 600.0)
>                  │ sortKey: OrdinalSortKey#1e145(order: 0.0)
>                  │
>                  └─SemanticsNode#3
>                    │ Rect.fromLTRB(0.0, 0.0, 800.0, 600.0)
>                    │ flags: scopesRoute
>                    │
>                    ├─SemanticsNode#4
>                    │ │ Rect.fromLTRB(0.0, 8.0, 80.0, 184.0)
>                    │ │ flags: hasImplicitScrolling
>                    │ │ scrollExtentMin: 0.0
>                    │ │ scrollPosition: 0.0
>                    │ │ scrollExtentMax: 0.0
>                    │ │
>                    │ ├─SemanticsNode#5
>                    │ │   Rect.fromLTRB(0.0, 0.0, 80.0, 44.0)
>                    │ │   actions: tap
>                    │ │   flags: isSelected, isFocusable
>                    │ │   label:
>                    │ │     "Abc
>                    │ │     Tab 1 of 4"
>                    │ │   textDirection: ltr
>                    │ │
>                    │ ├─SemanticsNode#6
>                    │ │   Rect.fromLTRB(0.0, 44.0, 80.0, 88.0)
>                    │ │   actions: tap
>                    │ │   flags: isFocusable
>                    │ │   label:
>                    │ │     "Def
>                    │ │     Tab 2 of 4"
>                    │ │   textDirection: ltr
>                    │ │
>                    │ ├─SemanticsNode#7
>                    │ │   Rect.fromLTRB(0.0, 88.0, 80.0, 132.0)
>                    │ │   actions: tap
>                    │ │   flags: isFocusable
>                    │ │   label:
>                    │ │     "Ghi
>                    │ │     Tab 3 of 4"
>                    │ │   textDirection: ltr
>                    │ │
>                    │ └─SemanticsNode#8
>                    │     Rect.fromLTRB(0.0, 132.0, 80.0, 176.0)
>                    │     actions: tap
>                    │     flags: isFocusable
>                    │     label:
>                    │       "Jkl
>                    │       Tab 4 of 4"
>                    │     textDirection: ltr
>                    │
>                    └─SemanticsNode#9
>                        Rect.fromLTRB(80.0, 290.0, 800.0, 310.0)
>                        label: "body"
>                        textDirection: ltr
>           The semantics tree would have matched the following configuration:
>             TestSemantics.root(
>               children: <TestSemantics>[
>                 TestSemantics(
>                   id: 1,
>                   textDirection: TextDirection.ltr,
>                   children: <TestSemantics>[
>                     TestSemantics(
>                       id: 2,
>                       children: <TestSemantics>[
>                         TestSemantics(
>                           id: 3,
>                           flags: <SemanticsFlag>[SemanticsFlag.scopesRoute],
>                           children: <TestSemantics>[
>                             TestSemantics(
>                               id: 4,
>                               flags: <SemanticsFlag>[SemanticsFlag.hasImplicitScrolling],
>                               children: <TestSemantics>[
>                                 TestSemantics(
>                                   id: 5,
>                                   flags: <SemanticsFlag>[SemanticsFlag.isSelected,
> SemanticsFlag.isFocusable],
>                                   actions: <SemanticsAction>[SemanticsAction.tap],
>                                   label: 'Abc\nTab 1 of 4',
>                                   textDirection: TextDirection.ltr,
>                                 ),
>                                 TestSemantics(
>                                   id: 6,
>                                   flags: <SemanticsFlag>[SemanticsFlag.isFocusable],
>                                   actions: <SemanticsAction>[SemanticsAction.tap],
>                                   label: 'Def\nTab 2 of 4',
>                                   textDirection: TextDirection.ltr,
>                                 ),
>                                 TestSemantics(
>                                   id: 7,
>                                   flags: <SemanticsFlag>[SemanticsFlag.isFocusable],
>                                   actions: <SemanticsAction>[SemanticsAction.tap],
>                                   label: 'Ghi\nTab 3 of 4',
>                                   textDirection: TextDirection.ltr,
>                                 ),
>                                 TestSemantics(
>                                   id: 8,
>                                   flags: <SemanticsFlag>[SemanticsFlag.isFocusable],
>                                   actions: <SemanticsAction>[SemanticsAction.tap],
>                                   label: 'Jkl\nTab 4 of 4',
>                                   textDirection: TextDirection.ltr,
>                                 ),
>                               ],
>                             ),
>                             TestSemantics(
>                               id: 9,
>                               label: 'body',
>                               textDirection: TextDirection.ltr,
>                             ),
>                           ],
>                         ),
>                       ],
>                     ),
>                   ],
>                 ),
>               ],
>             )
> 
> When the exception was thrown, this was the stack:
> #4      main.<anonymous closure> (file:///b/s/w/ir/x/w/flutter/packages/flutter/test/material/navigation_rail_test.dart:2276:5)
> <asynchronous suspension>
> #5      testWidgetsWithLeakTracking.wrappedCallBack (package:leak_tracker_flutter_testing/src/test_widgets.dart:108:5)
> <asynchronous suspension>
> #6      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:168:15)
> <asynchronous suspension>
> #7      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:1017:5)
> <asynchronous suspension>
> <asynchronous suspension>
> (elided one frame from package:stack_trace)
> 
> This was caught by the test expectation on the following line:
>   file:///b/s/w/ir/x/w/flutter/packages/flutter/test/material/navigation_rail_test.dart line 2276
> The test description was:
>   Semantics - labelType=[none]
> ════════════════════════════════════════════════════════════════════════════════════════════════════
> 
> 07:58 +8848 ~22 -1: /b/s/w/ir/x/w/flutter/packages/flutter/test/material/navigation_rail_test.dart: Semantics - labelType=[none] [E]                                                                   
>   Test failed. See exception logs above.
>   The test description was: Semantics - labelType=[none]
> ```
> 
> (from https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8761948290894190433/+/u/run_test.dart_for_framework_tests_shard_and_subshard_libraries/test_stdout).
> 
> That one will likely need to get fixed. If you see other unrelated test failures, try rebasing this PR with the latest master and see if they go away. The tree is currently green, so there are no unrelated existing failures.

I checked and tested this issue several times, I even returned the codes before I changed them, but I get the same error again! I don't think it is related to this PR

**Comment by goderbauer** [comment]
In that output of the test there is a new scrollable semantics node that wasn't there before if you compare expected and actual:

```
                   ├─SemanticsNode#4
                   │ │ Rect.fromLTRB(0.0, 8.0, 80.0, 184.0)
                   │ │ flags: hasImplicitScrolling
                   │ │ scrollExtentMin: 0.0
                   │ │ scrollPosition: 0.0
                   │ │ scrollExtentMax: 0.0
```

This is node is almost guaranteed produced by the SingleChildScrollView added in this PR. So, in order to get the tests passing, this test will have to be adjusted to account for this extra node.

**Comment by mbfakourii** [comment]
> In that output of the test there is a new scrollable semantics node that wasn't there before if you compare expected and actual:
> 
> ```
>                    ├─SemanticsNode#4
>                    │ │ Rect.fromLTRB(0.0, 8.0, 80.0, 184.0)
>                    │ │ flags: hasImplicitScrolling
>                    │ │ scrollExtentMin: 0.0
>                    │ │ scrollPosition: 0.0
>                    │ │ scrollExtentMax: 0.0
> ```
> 
> This is node is almost guaranteed produced by the SingleChildScrollView added in this PR. So, in order to get the tests passing, this test will have to be adjusted to account for this extra node.

Fixed, can you please check again?

**Comment by goderbauer** [comment]
Can you resolve the merge conflict, please? Thanks.

**Comment by mbfakourii** [comment]
> Can you resolve the merge conflict, please? Thanks.

Fixed.

File: packages/flutter/test/material/navigation_rail_test.dart [hunk 1]
```diff
@@ -5423,43 +5479,56 @@ TestSemantics _expectedSemantics() {
   return TestSemantics.root(
     children: <TestSemantics>[
       TestSemantics(
+        id: 1,
```
Comment (goderbauer) [inline packages/flutter/test/material/navigation_rail_test.dart:5482]: Can you please omit the IDs from this test like it was before? They are not relevant for the test and just make it more brittle if the ID assignment algorithm changes.

Comment (mbfakourii) [reply packages/flutter/test/material/navigation_rail_test.dart]: Deleted.

**Review by goderbauer — APPROVED** [review]
LGTM

**Review by QuncCccccc — APPROVED** [review]
LGTM! Nice improvement:) Thanks for the contribution!

**Comment by goderbauer** [comment]
Reason for revert: https://github.com/flutter/flutter/issues/143061, this is breaking some existing use cases, see also https://github.com/flutter/samples/pull/2157. If we try this again, we need to add this in less breaking way.

**Pull request merged**